### ISSUE TITLE
lambda.Function does not work well with Pulumi FileArchive

### ISSUE THREAD

**Issue by Jimmy89** [pulumi/pulumi-aws-native#743]
### What happened?

When using pulumi.asset.AssetArchive with the pulumi.asset.FileArchive the FileArchive argument is found incorrect.

Trying out the exact same code with aws-classic and there is no issue at all.

### Steps to reproduce

```
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws-native";
import * as awsClassic from "@pulumi/aws";

// For simplicity I added both examples (working and non-working)
new awsClassic.lambda.Function("working", {
  code: new pulumi.asset.AssetArchive({
    ".": new pulumi.asset.FileArchive("./lambdas/aFolder/"),
  }),
  runtime: "nodejs18.x",
  handler: "index.handler",
  role: basicLambdaRunRole.arn,
});

// This goes wrong
new aws.lambda.Function("error", {
  code: new pulumi.asset.AssetArchive({
    ".": new pulumi.asset.FileArchive("./lambdas/aFolder/"), <-- this lines triggers an error with `aws-native` package, but not with `aws-classic` package
  }),
  runtime: "nodejs18.x",
  handler: "index.handler",
  role: basicLambdaRunRole.arn,
});
```

### Expected Behavior

That the pulumi asset archive is being accepted by the aws-native package as well.

### Actual Behavior

```
 aws-native:lambda:Function (NAME):
    error: unexpected Asset Archive property value for "."
```

### Output of `pulumi about`

```
Dependencies:
NAME                VERSION
@pulumi/aws         5.23.0
@pulumi/aws-native  0.44.0
@pulumi/pulumi      3.49.0
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by squaremo** [comment]
Thanks for posting this. The `code` field in aws-native/lambda.Function takes completely different values to the analogous field in AWS classic: https://www.pulumi.com/registry/packages/aws-native/api-docs/lambda/function/#functioncode
There's a bug about making AWS Native work at least similarly: https://github.com/pulumi/pulumi-aws-native/issues/124

Sorry for the confusion! As a workaround, you can upload your code to an S3 bucket and refer to that, or (if it's just a single file) you might be able to read a file in a provide the contents in the field `zipFile`.

I think this can be closed as a duplicate of #124, but I'll wait to see your response @Jimmy89.

**Comment by Jimmy89** [comment]
@squaremo Thank you for your comment. It's indeed a duplicate of #124. I will close this issue
